.circleci/config.yml

```
@@ -217,121 +173,133 @@ commands:
-      - run:
-          name: Run Python unit tests
-          command: |
-            conda activate kedro-viz
-            rm -r build, package/dist, package/kedro_viz/html, pip-wheel-metadata, package/kedro_viz.egg-info -ErrorAction Ignore
-            npm run build
-            cp -R build package/kedro_viz/html
-            pytest package/tests --no-cov
-      - run:
-          name: Run e2e tests
-          command: |
-            conda activate kedro-viz
-            cd package
-            behave
-
-jobs:
-  build_37:
-    executor: circleci_machine_37
-    steps: [build]
+      - run_win_e2e_tests
 
-  build_38:
-    executor: circleci_machine_38
-    steps: [build]
-
-  build_39:
-    executor: circleci_machine_39
-    steps: [build]
-
-  build_310:
-    executor: circleci_machine_310
-    steps: [build]
-
-  deploy_demo:
-    executor: demo_executor
-    steps: [deploy_demo]
+  unit_tests:
+    parameters:
+      python_version:
+        type: string
+    executor:
+      name: docker
+      python_version: <<parameters.python_version>>
+    steps:
+      - checkout 
+      - setup_python_env
+      - install_node_dependencies
+      - run_python_tests
 
-  # Windows-related jobs
-  win_build_37:
-    executor: win/default
+  win_unit_tests:
     parameters:
       python_version:
         type: string
-    working_directory: ~/repo
+    executor: win/default
     steps:
-      - checkout
-      - win_build:
+      - checkout 
+      - win_setup_conda:
           python_version: <<parameters.python_version>>
+      - win_setup_requirements
+      - install_node_dependencies
+      - run_win_python_tests
 
-  win_build_38:
-    executor: win/default
+  
+  lint:
     parameters:
       python_version:
         type: string
-    working_directory: ~/repo
+    executor:
+      name: docker
+      python_version: <<parameters.python_version>>
     steps:
-      - checkout
-      - win_build:
-          python_version: <<parameters.python_version>>
+      - setup_python_env
+      - run_python_format_and_lint
 
-  win_build_39:
-    executor: win/default
+  
+  javascript_lint_and_tests:
     parameters:
       python_version:
         type: string
-    working_directory: ~/repo
+    executor:
+      name: docker
+      python_version: <<parameters.python_version>>
+    description: Run javascript commands and tests
     steps:
-      - checkout
-      - win_build:
-          python_version: <<parameters.python_version>>
+      - checkout 
+      - install_node_dependencies
+      - test_lib_transpilation
+      - test_lib_import
+      - run_eslint
+      - run_javascript_tests
 
-  win_build_310:
-    executor: win/default
+  build_package: 
     parameters:
       python_version:
         type: string
-    working_directory: ~/repo
+    executor:
+        name: docker
+        python_version: <<parameters.python_version>>
```

```suggestion
        python_version: "3.8"
```
Since this is only used with one Python version, let's not bother with the matrix notation for it. Similarly for any other jobs that are only invoked with one Python version.